clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -898,6 +837,30 @@ private HeartbeatRequestManager createHeartbeatRequestManager(
                 heartbeatState,
                 heartbeatRequestState,
                 backgroundEventHandler,
-                metrics);
+                new Metrics());
+    }
+
+    private void mockJoiningMemberData(String instanceId) {
+        when(membershipManager.groupInstanceId()).thenReturn(Optional.ofNullable(instanceId));
+        when(membershipManager.memberId()).thenReturn("");
+        when(membershipManager.memberEpoch()).thenReturn(0);
+        when(membershipManager.groupId()).thenReturn(DEFAULT_GROUP_ID);
+        when(membershipManager.currentAssignment()).thenReturn(LocalAssignment.NONE);
+        when(membershipManager.serverAssignor()).thenReturn(Optional.of(DEFAULT_REMOTE_ASSIGNOR));
+    }
+
+    private void mockFencedToJoiningMemberData() {
```

Comment:
nit: `mockRejoiningMemberData` is maybe better (fencing is just only one of the ways to get to rejoining, not specifically used here)